CMakeLists.txt

```diff
@@ -0,0 +1,630 @@
+    src/loaders/prowizard/ptktable.c
+    src/loaders/prowizard/skyt.c
+    src/loaders/prowizard/starpack.c
+    src/loaders/prowizard/tdd.c
+    src/loaders/prowizard/theplayer.c
+    src/loaders/prowizard/titanics.c
+    src/loaders/prowizard/tp1.c
+    src/loaders/prowizard/tp3.c
+    src/loaders/prowizard/tuning.c
+    src/loaders/prowizard/unic2.c
+    src/loaders/prowizard/unic.c
+    src/loaders/prowizard/wn.c
+    src/loaders/prowizard/xann.c
+    src/loaders/prowizard/zen.c
+)
+
+set(LIBXMP_SRC_LIST_DEPACKERS
+    src/depackers/arcfs.c
+    src/depackers/bunzip2.c
+    src/depackers/crc32.c
+    src/depackers/depacker.c
+    src/depackers/gunzip.c
+    src/depackers/inflate.c
+    src/depackers/mmcmp.c
+    src/depackers/muse.c
+    src/depackers/oxm.c
+    src/depackers/ppdepack.c
+    src/depackers/readlzw.c
+    src/depackers/readrle.c
+    src/depackers/s404_dec.c
+    src/depackers/unarc.c
+    src/depackers/uncompress.c
+    src/depackers/unlha.c
+    src/depackers/unlzx.c
+    src/depackers/unsqsh.c
+    src/depackers/unxz.c
+    src/depackers/unzip.c
+    src/depackers/vorbis.c
+    src/depackers/xfnmatch.c
+    src/depackers/xz_dec_lzma2.c
+    src/depackers/xz_dec_stream.c
+)
+
+if(AMIGA)
+    list(APPEND LIBXMP_SRC_LIST_DEPACKERS
+        src/depackers/xfd.c
+        src/depackers/xfd_link.c
+    )
+endif()
+
+if(WIN32)
+    list(APPEND LIBXMP_SRC_COMMON
+        src/win32.c
+        src/win32/ptpopen.c
+    )
+    if(MSVC AND CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+        list(APPEND LIBXMP_SRC_COMMON
+            src/win32/debug.c
+        )
+    endif()
+endif()
+
+set(LIBXMP_DEFINES)
+
+if(DISABLE_IT)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_CORE_DISABLE_IT)
+endif()
+
+if(LIBXMP_DISABLE_DEPACKERS)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_NO_DEPACKERS)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL_ONLY ${LIBXMP_SRC_LIST_DEPACKERS})
+endif()
+
+if(LIBXMP_DISABLE_PROWIZARD)
+    list(APPEND LIBXMP_DEFINES -DLIBXMP_NO_PROWIZARD)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL_ONLY ${LIBXMP_SRC_LIST_PROWIZARD})
+endif()
+
+list(APPEND LIBXMP_SRC_LITE ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_LITE_ONLY} ${LIBXMP_SRC_COMMON})
+list(APPEND LIBXMP_SRC_FULL ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_FULL_ONLY} ${LIBXMP_SRC_LIST_FULL} ${LIBXMP_SRC_COMMON})
+
+set(XMP_INSTALLS)
+
+# ============= FULL =============
+if(LIBXMP_BUILD_FULL AND BUILD_STATIC)
+    add_library(xmp_static STATIC ${LIBXMP_SRC_FULL})
+    list(APPEND XMP_INSTALLS xmp_static)
+    set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp C_STANDARD 99)
+
+    if(WIN32 AND BUILD_SHARED)
+        set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp-static)
+    else()
+        set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp)
+    endif()
+
+    target_compile_definitions(xmp_static PRIVATE ${LIBXMP_DEFINES} -DBUILDING_STATIC)
+
+    if(MSVC)
+        target_compile_definitions(xmp_static PRIVATE -DPATH_MAX=260 -DCLIB_DECL=__stdcall)
```

Comment:
What is the purpose of `__stdcall` here? Don't.

Reply:
I don't remind why I added this, let me check the history...